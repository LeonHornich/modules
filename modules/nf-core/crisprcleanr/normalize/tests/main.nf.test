nextflow_process {

    name "Test Process CRISPRCLEANR_NORMALIZE"
    script "../main.nf"
    process "CRISPRCLEANR_NORMALIZE"

    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "crisprcleanr"
    tag "crisprcleanr/normalize"


    test("homo_sapiens - [crispr_functional_counts, crispr_functional_library]") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['crispr_functional_counts'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['crispr_functional_library'], checkIfExists: true)
                ]
                input[1] = 30
                input[2] = 5
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.norm_count_file).match("norm_count_file") },
                { assert snapshot(process.out.versions).match("versions") }
            )

        }

    }

    test("homo_sapiens - [crispr_functional_counts, crispr_functional_library] - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['genome']['crispr_functional_counts'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['genome']['crispr_functional_library'], checkIfExists: true)
                ]
                input[1] = 30
                input[2] = 5
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )

        }

    }

}