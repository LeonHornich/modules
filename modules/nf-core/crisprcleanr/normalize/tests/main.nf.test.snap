{
    "homo_sapiens - [crispr_functional_counts, crispr_functional_library]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_norm_table.tsv:md5,ba36e7e31a511ba4dadc888ea13f30e7"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e954dc2bfa394dae2e55be2b1c16ae31"
                ],
                "norm_count_file": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_norm_table.tsv:md5,ba36e7e31a511ba4dadc888ea13f30e7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e954dc2bfa394dae2e55be2b1c16ae31"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T07:15:50.114438385"
    },
    "homo_sapiens - [crispr_functional_counts, crispr_functional_library] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_norm_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d927535d45027d86be9f9348d5bca474"
                ],
                "norm_count_file": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_norm_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d927535d45027d86be9f9348d5bca474"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T07:16:03.002248203"
    }
}